app/jobs/work_index_job.rb

```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+class WorkIndexJob < Hyrax::ApplicationJob
+  def perform(work)
+    work.update_index
```

Comment:
Perhaps same thing here

```suggestion
    return unless work

    Hyrax.index_adapter.save(resource: work)
```